Related to #11696

ojfd opened the pull request:
Adds Baofeng BF-9700 + Stun entries in Browser. Removes 'Scan' checkbox for BF-190x and BF-9700 not present in OEM software. Addresses a few E722 warnings (not 100% sure about them).

ojfd committed:
Add files via upload

ojfd committed:
Related to #11696

Adds Baofeng BF-9700. Removes 'Scan' checkbox for BF-190x and BF-9700 not present in OEM software. Addresses a few E722 warnings (not 100% sure about them).

ojfd committed:
Related to #11696

Adds Baofeng BF-9700 + Stun entries in Browser. Removes 'Scan' checkbox for BF-190x and BF-9700 not present in OEM software. Addresses a few E722 warnings.

ojfd committed:
Related to #11696 v2 to test

ojfd committed:
Related to #11696 v3 to test

ojfd committed:
Delete h777.py

ojfd committed:
th_uv88: adds Dual Watch + other settings for Retevis P2/P62

Related to #11864. Fixes swapped Last and Main channels in Priority Transmit for RA89/P62/P2.
Adds Mic Level for RA89. Adds freq. band reporting in 'Browser' tab.

ojfd committed:
th_uv88: to test for Python 3.x drivers

ojfd committed:
Update th_uv88.py

ojfd committed:
Cosmetic changes.

Renumbering 'unkXX' in sequential order. Added 'Eliminate Squelch Tail When No Ctc/Dcs Signaling' entry to be seen in 'Browser'

ojfd committed:
Update th_uv88.py

ojfd committed:
Update th_uv88.py

Put two underscores back

kk7ds commented:
Almost none of the contributors to CHIRP are professional programmers, FWIW :)

And no, I'm talking to you like the obviously intelligent person that you are and trying to point you at the steps we all go through to get stuff into the tree in an organized way. These may seem foreign at first, but they're very doable. You mentioned (what I deduced was) this PR in a comment earlier today in a way that made it seem like you were expecting it to be merged. I had seen it marked as draft before and was still in a state needing to be cleaned up so I figured maybe I'd circle back here and nudge you towards the issues keeping it from being merged in case it wasn't obvious.

I'll pull this down and reformat it the way it should be, but I'll want you to look it over again since I don't have any of these radios to test with.

kk7ds commented on chirp/drivers/th_uv88.py:
```diff
@@ -871,30 +903,53 @@ def get_settings(self):
+                   '4.0S', '4.5S', '5.0S']
+        rx = RadioSettingValueList(options, current_index=_settings.voxDelay)
+        rset = RadioSetting("basicsettings.voxDelay", "VOX Delay", rx)
+        basic.append(rset)
```
Help me out... this is just moved from below but otherwise unchanged? Is there some reason it was moved or just accidental?

ojfd replied:
Yes, purely for cosmetic reasons, so that Vox Level and Vox Delay appear in one place in GUI

kk7ds replied:
Okay cool, so this change should be one commit on its own so it's clear that it's just moving (and not changing) anything.

ojfd replied:
It does. But, just FYI, in OEM software all these functions are in one, s.c. "Basic" window

kk7ds commented on chirp/drivers/th_uv88.py:
```diff
@@ -871,30 +903,53 @@ def get_settings(self):
+        rx = RadioSettingValueBoolean(_settings.dualWait)
+        rset = RadioSetting("basicsettings.dualWait",
+                            "Dual Wait/Standby", rx)
+        basic.append(rset)
```
This adds the dual wait setting for all models, where it was restricted to only ones with LCDs before. Is this intentional? I have h777-based radios without displays that I'm sure don't have dual wait since they only have a way to select a single channel.

ojfd replied:
h777 based radios in uv88 file?

kk7ds commented on chirp/drivers/th_uv88.py:
```diff
@@ -871,30 +903,53 @@ def get_settings(self):
+                    [(str(x + 1), x) for x in range(200)], _settings.dwchan)
+                rs = RadioSetting('basicsettings.dwchan',
+                                  'Dual Wait Channel', rsv)
+                basic.append(rs)
```
I'm guessing these non-display radios have a dedicated "always dual wait for channel X" function? Mine aren't in this list, which means they'll get a dual-wait setting in settings, but won't get this extra setting either. I think the logic should be "if LCD, always add dual-wait. If not LCD but `dwchan in _settings` then add dual-wait and dual-wait-chanel" right?

kk7ds replied:
This and the question above are still relevant to make sure you intended to change this in this way, however my confidence about whether it's correct or not is irrelevant since I don't know anything about these radios.

@kc9hi can you review the changes being made to this driver to make sure they jive with your understanding of them all?

ojfd replied:
Nah, it's just two models. Almost identical. One with display - Retevis RA89/TYT TH-UV99, the other one without - Retevis P62/P2/ TYT TC29.
I've double and triple checked all functions on my radios, everything is working as expected. I have a park of all of these. Found a few firmware bugs too along the way.
I keep filing on this driver and it would be extremely cool if @KC9HI could revisit it. I have a bunch of question to ask

kk7ds commented on chirp/drivers/th_uv88.py:
```diff
@@ -871,30 +903,53 @@ def get_settings(self):
+            options = ["Main Channel", "Last Channel"]
+        else:
+            options = ["Main Channel", "Last Channel",
+                       "Active Channel with Lockout Time"]
```
Are you sure all h777-based radios without LCD support this extra setting? Or do you want to add that only for the subset of models you know have it?

kk7ds reviewed (commented):
A couple of comments of things that should change, and questions about the changes being made here so I can split and cleanup the the commit history as noted.

Thanks!

ojfd commented on chirp/drivers/h777.py:
```diff
@@ -516,9 +527,10 @@ def get_settings(self):
-        rs = RadioSetting("scan", "Scan",
-                          RadioSettingValueBoolean(_settings.scan))
-        basic.append(rs)
+        if self.PROGRAM_CMD == b'PROGRAM':
```
I did as best as I could. Not the most elegant way, I know. Quick and dirty. I just wanted to get rid of "Scan" for radios that do not support it.

ojfd committed:
Update th_uv88.py

ojfd committed:
Removed two arrays of 1

ojfd committed:
Merge branch 'kk7ds:master' into master

ojfd committed:
Merge branch 'kk7ds:master' into master

ojfd committed:
Merge branch 'kk7ds:master' into master

ojfd committed:
Merge branch 'kk7ds:master' into master

ojfd committed:
Merge branch 'kk7ds:master' into master

ojfd committed:
Merge branch 'kk7ds:master' into master

ojfd committed:
Merge branch 'kk7ds:master' into master

ojfd committed:
Add new official firmware for v2 radios

From about august Quansheng is shipping all UV-K5 variants with new controller chip and new firmware. Radios with previous controller chip are no longer being manufactured. Numbering of new firmware starts with 1.

ojfd committed:
Merge branch 'master' into master

ojfd committed:
Merge branch 'kk7ds:master' into master

ojfd committed:
Merge branch 'kk7ds:master' into master
